FAL-2260 Prepare opencraft release branch for lilac.2

samuelallan72 opened the pull request:
[FAL-2260](https://tasks.opencraft.com/browse/FAL-2260)

Port the following PRs / commits from koa to lilac:

- https://github.com/open-craft/configuration/pull/162
- https://github.com/open-craft/configuration/pull/164
- https://github.com/open-craft/configuration/pull/159
- https://github.com/open-craft/configuration/pull/167
- https://github.com/open-craft/configuration/pull/169
- https://github.com/edx/configuration/commit/4df1323d8906675147acb7c448246808e334e3fd
- https://github.com/open-craft/configuration/pull/173

These are skipped because they have already been upstreamed in lilac :tada: :

- https://github.com/open-craft/configuration/pull/170

And this isn't required any more (see comments below):

- https://github.com/open-craft/configuration/pull/140

**Test instructions**:

For each patch from https://github.com/edx/configuration/compare/open-release/koa.master...open-craft:opencraft-release/koa.3 , verify that it has either been upstreamed or ported cleanly.

Launch a lilac instance with ocim, using this configuration branch, and verify everything is deployed successfully. (wip - deploying on https://manage.opencraft.com/instance/29580/ )

**Notes / Concerns**:

- [x] Not sure about https://github.com/open-craft/configuration/pull/175/commits/93ba25ec09f6f071e54a984c1f26721dc0bf1d2c - we don't need to use our fork of tubular any more, since upstream has merged https://github.com/edx/tubular/pull/532 .  The upstream fix is in lilac.master though, not lilac.2.  I'd recommend modifying that commit to pin to edx/tubular@open-release/lilac.master , but not sure if that's what we want to do here.  I'm not sure why we're pinning it here, and not in private extra-configuration. CC @lgp171188 
- [x] Also not sure about https://github.com/open-craft/configuration/pull/140 - upstream had changed the import paths CC @kaizoku
- [x] I couldn't find an associated PR for https://github.com/edx/configuration/commit/4df1323d8906675147acb7c448246808e334e3fd CC @pkulkark could you check this?

0x29a committed:
BB-3669 feat: ansible task to install extra edx django service requirements

https://github.com/open-craft/configuration/pull/162

(cherry picked from commit 9daa748f9d1195220fa6d751b6720503b17855de)

Commit:
[SE-4419] Add script to generate the daily uptime reports (#164)

https://github.com/open-craft/configuration/pull/164

(cherry picked from commit 590f3529c2e8ff3f2c68965ad2af2e66e7f1a271)

pkulkark committed:
[SE-4364] Update reindex_courses command to use --all option and move it into forum role (#159)

https://github.com/open-craft/configuration/pull/159

(cherry picked from commit 066fbf5d3d1bb21e5788510fe377471395741cad)

Commit:
[SE-4481]: Upload edx-services logs to s3 bucket  (#167)

This PR add the flags to sync the edx services logs to s3 bucket.

(cherry picked from commit eb402aa29492690559a86b37944e5c587c9f6042)

Commit:
Merge pull request #169 from open-craft/joaocabrita/bb_4342-credentials_extra_requirements_koa

feature: Allow installing extra requirements when using the `credentials` role
(cherry picked from commit 7ce36c7953f505246cd0cc4999d284270a190eea)

pkulkark committed:
fix: put edx service logs sync behind flag

(cherry picked from commit 4df1323d8906675147acb7c448246808e334e3fd)

0x29a committed:
Merge pull request #173 from open-craft/guruprasad/FAL-2409-tubular-pygithub-fix

[FAL-2409] Use the tubular pygithub fix in our common branch

(cherry picked from commit a957ce3491cfd95b6b8759d2c07830cebd93fdc7)

michaelbukachi commented on playbooks/retire_host.yml:
```diff
@@ -38,7 +38,13 @@
+    - name: Sync edX services logs again
+      command: "/edx/bin/send-logs-to-object-store -d {{ COMMON_LOG_DIR }}{{ item.src }} -b {{ COMMON_OBJECT_STORE_LOG_SYNC_BUCKET }} -p {{ item.dest }}"
+      with_items:
+        - { src: '/lms/',  dest: '{{ COMMON_OBJECT_STORE_EDX_LOG_SYNC_PREFIX }}lms/' }
```
@swalladge This line is missing from this [PR](https://github.com/open-craft/configuration/pull/167/files):
```
        - { src: '/tracking/',  dest: '{{ COMMON_OBJECT_STORE_LOG_SYNC_PREFIX }}' }
```

samuelallan72 replied:
@michaelbukachi interesting - looks like this line was removed in this commit: https://github.com/open-craft/configuration/pull/175/commits/3451ed1e48d2ec9ffad0843ffc7e7a13f9ab6222  I can't link that commit to a PR though - however it does look like a follow up commit to the PR you linked. CC @pkulkark

samuelallan72 replied:
(So I think this is fine, but it would be nice to figure out which ticket/PR that last commit is linked to.)

michaelbukachi replied:
@swalladge noted.

michaelbukachi commented:
I don't see the changes from [[SE-4304] [SE-4495] feat: add celerybeat configuration](https://github.com/open-craft/configuration/pull/170/files).

samuelallan72 commented:
@michaelbukachi 

> I don't see the changes from [SE-4304] [SE-4495] feat: add celerybeat configuration.

This is this commit: https://github.com/open-craft/configuration/pull/175/commits/dd2f2ac377f4f542da336019e5bee6147bf29568  The diff shows less changes than the original commit, because most of it has been upstreamed.

CC @gabor-boros do we need the 'install simple-beat to control celerybeat processes' task? Will it be upstreamed?

michaelbukachi commented:
@swalladge looks good.

gabor-boros commented:
@swalladge I'm a bit confused by your question. The changes are upstreamed as part of

* https://github.com/edx/configuration/pull/6455 - Lilac
* https://github.com/edx/configuration/pull/6456 - Koa

Am I missing something here?

samuelallan72 commented:
@gabor-boros 

> I'm a bit confused by your question. The changes are upstreamed as part of ...

> Am I missing something here?

aaah sorry, I was the one missing something. :P  Yep it's all been upstreamed; git was playing tricks on me when I checked the diff and applied the commit though. :man_facepalming:

gabor-boros commented:
@swalladge I believe some PRs/commits are missing. Correct me if I overlooked it, but OCIM now can provision instances that are using multi tenant redis. Now, the celery upgrade is not upstreaming yet as I'm still fighting with the ticket in the current sprint.

Taking that into account, shouldn't we port that to opencraft lilac as well? Since it is something that will never end up on edx/lilac but we want to use it, I guess we should cherry-pick that too.

Cc: @pomegranited

samuelallan72 commented:
@gabor-boros 

> I believe some PRs/commits are missing. Correct me if I overlooked it, but OCIM now can provision instances that are using multi tenant redis. Now, the celery upgrade is not upstreaming yet as I'm still fighting with the ticket in the current sprint.

> Taking that into account, shouldn't we port that to opencraft lilac as well?

Sure yes definitely.  Are these commits/PRs you're talking about on edx/configuration and in koa?  I didn't see any other commits related to redis/celery.  If they're in edx/configuration master, then we can backport to our lilac branch after this commit is merged (ie. after we actually cut our lilac release).

gabor-boros commented:
> @gabor-boros
> 
> > I believe some PRs/commits are missing. Correct me if I overlooked it, but OCIM now can provision instances that are using multi tenant redis. Now, the celery upgrade is not upstreaming yet as I'm still fighting with the ticket in the current sprint.
> 
> > Taking that into account, shouldn't we port that to opencraft lilac as well?
> 
> Sure yes definitely. Are these commits/PRs you're talking about on edx/configuration and in koa? I didn't see any other commits related to redis/celery. If they're in edx/configuration master, then we can backport to our lilac branch after this commit is merged (ie. after we actually cut our lilac release).

Well, it seems I remembered wrongly. Sorry about the confusion!

pkulkark commented:
@swalladge 
> I couldn't find an associated PR for edx@4df1323 CC @pkulkark could you check this?

I believe that was a backport of this upstream PR: https://github.com/edx/configuration/pull/6465.

samuelallan72 commented:
@pkulkark 

> I believe that was a backport of this upstream PR: edx#6465.

Ok.  Was there no corresponding PR for the backport?

kaizoku commented:
@swalladge, for #140 , the AppleIdAuth backend is in lilac now, so that particular PR shouldn't be necessary.
I do see they noted that [the path changed](https://github.com/edx/edx-platform/pull/24623/files#r560484081), which is [reflected in `lms/envs/production.py`](https://github.com/open-craft/edx-platform/blob/b509663042d17be270f863c7e827b8df0e954478/lms/envs/production.py#L604), so we'll need to updated those paths for `EDXAPP_THIRD_PARTY_AUTH_BACKENDS` in `playbooks/roles/edxapp/defaults/main.yml` too.

samuelallan72 commented:
@kaizoku 

> the AppleIdAuth backend is in lilac now, so that particular PR shouldn't be necessary.

Awesome! :)


> I do see they noted that the path changed, which is reflected in lms/envs/production.py, so we'll need to updated those paths for EDXAPP_THIRD_PARTY_AUTH_BACKENDS in playbooks/roles/edxapp/defaults/main.yml too.

Looks like that's already been done, so that's all good now.

samuelallan72 commented on playbooks/openedx_native.yml:
```diff
@@ -67,6 +67,15 @@
+        DOMAIN: "{{ COMMON_GIT_MIRROR }}"
+        PATH: "open-craft"
+        REPO: "tubular.git"
+        VERSION: "opencraft-release/koa.3"
```
@pomegranited what are your thoughts here?  We can't use open-release/lilac.2 for tubular because https://github.com/edx/tubular/pull/531 didn't make it to lilac.2, but we should default to a lilac release here to avoid issues.  Maybe we can default to open-release/lilac.master here?

pomegranited replied:
@swalladge That sounds safe -- tubular isn't changed very often, so we're not risking much by using a potentially-shifting lilac.master target.

samuelallan72 replied:
:+1:  I updated in https://github.com/open-craft/configuration/pull/175/commits/06fb8f4f4921601695586f080c2a6e68045c1602 - would you mind sanity checking this?

I'm also deploying a [new sandbox instance](https://manage.opencraft.com/instance/29580/edx-appserver/21078/) for testing.  CC @michaelbukachi if you'd like to make another review pass now that there has been some changes.

michaelbukachi replied:
On it.

pomegranited replied:
@swalladge 06fb8f4 looks good 👍 Thanks for simplifying that drift too.

@michaelbukachi  To test this, you can check the sandbox server for retirement-related errors in `/edx/var/log/**/*.log`, and verify that you're able to create and delete a user successfully.

samuelallan72 committed:
Pin retirement service (tubular) to lilac.master by default

Previously, we pinned to an opencraft release as a quick fix,
but that quick fix is now merged to lilac upstream.
(The fix being https://github.com/edx/tubular/pull/531 )
It didn't make it to lilac.2, so we need to pin to lilac.master.

michaelbukachi reviewed (approved):
LG

pomegranited commented:
@swalladge  I don't see your change from https://github.com/edx/configuration/pull/6462 in this branch? It was partially backported to our koa.3 release branch by https://github.com/open-craft/configuration/pull/176 but that port unfortunately doesn't include the `EDXAPP_ENABLE_MONGODB_INDEXES` and `FORUM_ENABLE_MONGODB_INDEXES` gates that your upstream PR included, creating some code drift.

I think https://github.com/edx/configuration/pull/6462 should be ported to this branch, and the `*_ENABLE_MONGODB_INDEXES` flags enabled in Ocim. Any client sites that want to use them will have to enable them too, CC @Agrendalath 

I will include this in the scope of [FAL-2543](https://tasks.opencraft.com/browse/FAL-2543), which is semi-related.

samuelallan72 commented:
@pomegranited 

> I don't see your change from edx#6462 in this branch? 

Yeah they wouldn't be unless we already had backported it to our koa branch, or upstream had included it in lilac.2...

> I think edx#6462 should be ported to this branch, and the *_ENABLE_MONGODB_INDEXES flags enabled in Ocim

Sounds like a good plan :+1:

> I will include this in the scope of FAL-2543, which is semi-related.

Thanks! :D

Agrendalath commented:
@pomegranited, @swalladge, huh, I had no idea we've created two tickets for this :facepalm:

samuelallan72 commented:
@Agrendalath oh haha I didn't even notice they were different tickets.  Looks like we forgot to backport to the client branch in the first ticket, and the second ticket didn't notice that we'd already implemented it upstream.
